src/broadcast/metadatabroadcast.cpp

```diff
@@ -0,0 +1,27 @@
+#include "metadatabroadcast.h"
+#include "mixer/playerinfo.h"
+
+MetadataBroadcaster::MetadataBroadcaster(TrackTimers::ElapsedTimer *timer) :
+m_pElapsedTimer(timer)
+{
+
+}
+
+void MetadataBroadcaster::slotReadyToBeScrobbled(TrackPointer pTrack) {
+
+}
+
+void MetadataBroadcaster::slotNowListening(TrackPointer pTrack) {
+    for (auto &service : m_scrobblingServices) {
```

Comment:
This is essentially a manually implemented synchronous pub/sub communication. Why don't you use Qt signal/slots for this purpose with all their thread-safety guarantees?

Reply:
I didn't use slots because I control when to call the services and I own the metadata broadcast object, which lives in the same thread as the scrobbling manager. Since I know when I will call the services and I don't need queued connections I didn't see any reason to use signals & slots.

Reply:
Send everything that you need at the receiver side by value, i.e. wrap it into a queueable meta-type.

Reply:
You mean all the track metadata so the services don't have to store a pointer? Like a DTO?